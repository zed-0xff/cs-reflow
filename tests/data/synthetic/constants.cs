partial class C1 {
    public const int Z = 2+2;
}

partial class C1 {
    public readonly static int X = 1; // can still be modified via reflection
    public const int Y = 2;

    void fun() {
        if (X == 1)
        {
            maybe_called();
        }

        if (Y != 2)
        {
            never_called();
        }

        if (Z == 4)
        {
            always_called();
        }
    }
}

class C2 {
    void fun2()
    {
        if (C1.Z == 4)
        {
            always_called();
        }
    }
}
