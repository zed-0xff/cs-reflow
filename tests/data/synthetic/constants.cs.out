void fun()
{
    if (X == 1)
    {
        maybe_called();
    }

    always_called();
}

void fun2()
{
    always_called();
}
